src/main/java/com/direwolf20/buildinggadgets/common/world_save/TemplateWorldSave.java

```diff
@@ -0,0 +1,77 @@
+package com.direwolf20.buildinggadgets.common.world_save;
+
+import com.direwolf20.buildinggadgets.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.schema.template.Template;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.INBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraft.world.storage.WorldSavedData;
+import net.minecraftforge.common.util.Constants.NBT;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public final class TemplateWorldSave extends WorldSavedData {
+    public static TemplateWorldSave getInstance(ServerWorld world) {
+        return null;//TODO: evaluate correct load Method
+    }
+
+    private static final String NAME = BuildingGadgets.MOD_ID + "_template_data";
+    private static final String KEY_TEMPLATE_LIST = "templates";
+    private static final String KEY_ID = "id";
+    private static final String KEY_DATA = "data";
+    private final Map<UUID, Template> loadedTemplates;
+
+    public TemplateWorldSave() {
+        super(NAME);
+        this.loadedTemplates = new HashMap<>();
```

Comment:
I don't think we need to store loadedTemplates

Reply:
Err, what?!? Of course we do.
Why would you keep the nbt data hanging around?

Reply:
What

Reply:
You just said we don't need loadedTemplates. The only other ways would be to keep the raw nbt data around and deserialize a Template from it everytime we want to use one. And that's just awful.